src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerSaveImageFunctionalTest.groovy

```diff
@@ -54,12 +62,130 @@ class DockerSaveImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
         file(CONTROL_SAVED_IMAGE).size() > file(COMPRESSED_IMAGE_FILE).size()
     }
 
-    static String pullImageTask() {
+    @Unroll
+    def "only saves specific Docker image [image: #image]"() {
+        buildFile << pullImageTask('pullImage', IMAGE_3_4)
+        buildFile << pullImageTask('pullImage2', IMAGE_3_3)
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.DockerSaveImage
+
+            task saveImage(type: DockerSaveImage) {
+                dependsOn tasks.withType(DockerPullImage)
+                image = "${image}"
+                destFile = file("${IMAGE_FILE}")
+            }
+        """
+
+        when:
+            build('saveImage')
+
+        then:
+            getRepoTags(IMAGE_FILE) == [image]
+
+        where:
+            image << [IMAGE_3_3, IMAGE_3_4]
+    }
+
+    def "saves all Docker images of specified base"() {
+        buildFile << pullImageTask('pullImage', IMAGE_3_4)
+        buildFile << pullImageTask('pullImage2', IMAGE_3_3)
+        buildFile << pullImageTask('pullImage3', IMAGE_3_2)
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.DockerSaveImage
+
+            task saveImage(type: DockerSaveImage) {
+                dependsOn tasks.withType(DockerPullImage)
+                image = "${IMAGE_BASE}"
+                destFile = file("${IMAGE_FILE}")
+            }
+        """
+
+        when:
+            build('saveImage')
+
+        then:
+            def repoTags = getRepoTags(IMAGE_FILE)
+            repoTags.containsAll(IMAGE_3_2, IMAGE_3_3, IMAGE_3_4)
+            repoTags.every { it.startsWith("${IMAGE_BASE}:") }
```

Comment:
I feel like you already tested this with the previous assertion. What I would probably do it is see if the size of `repoTags` is 3 and if the collection contains the expected images.

Reply:
Not really, that would make the test flaky.
If there are other alpine images in your local Docker daemon, they will be exported alongside as the command instructs to export all alpine tags.
So the last line ensures that as expected all exported tags are alpine tags and the line before that at least the three that are guaranteed by the test code are included, but there can be more.
Unless some code is added to delete all other alpine tags from the Docker daemon, but I didn't want to do that to anyone just running those tests.